File: server/src/main/java/com/fluffy/global/response/PageResponse.java
package com.fluffy.global.response;

import java.util.List;
import org.springframework.data.domain.Page;

public record PageResponse<T>(
        PageInfo pageInfo,
        List<T> content
) {
    public static <T> PageResponse<T> of(Page<T> page) {
        return new PageResponse<>(
                new PageInfo(
                        page.getNumber(),
                        page.getTotalPages(),
                        page.getTotalElements(),
                        page.hasNext(),
                        page.hasPrevious()
                ),
                page.getContent()
        );
    }
}

File: server/src/main/kotlin/com/fluffy/global/response/PageResponse.kt
package com.fluffy.global.response

import org.springframework.data.domain.Page

data class PageResponse<T>(
    val pageInfo: PageInfo,
    val content: List<T>
) {
    companion object {
        fun <T> of(page: Page<T>): PageResponse<T> {
            return PageResponse(
                PageInfo(
                    page.number,
                    page.totalPages,
                    page.totalElements,
                    page.hasNext(),
                    page.hasPrevious()
                ),
                page.content
            )
        }
    }
}
